### ISSUE TITLE
No attribute "device" when using Numpy backend

### ISSUE THREAD

**Issue by pfisterj** [tensorly/tensorly#545]
#### Describe the bug
When using Numpy backend, it seems that Tensorly tensors have no attribute "device".

#### Steps or Code to Reproduce
```python
import tensorly as tl

# works OK
tl.set_backend('pytorch')
A = tl.tensor((3,3),dtype=tl.float64,device="cpu")
print("torch|A:",A)
print("torch|A.dtype:",A.dtype)
print("torch|A.device:",A.device)

# raises an AttributeError: 'numpy.ndarray' object has no attribute 'device' 
# when printing A.device
tl.set_backend('numpy')
A = tl.tensor((3,3),dtype=tl.float64,device="cpu")
print("numpy|A:",A)
print("numpy|A.dtype:",A.dtype)
print("numpy|A.device:",A.device)
```

#### Expected behavior
It would be nice to have a "device" attribute set to "cpu" when Numpy is used as backend. The reason behind this is that sometimes we may want to create a new tensor, having e.g. similar 'device' and 'dtype' than another tensor. Even if a tensor with Numpy backend will obviously live on the cpu, this would prevent from having to add an if: else: block in the code.

Note also that ```A = tl.tensor((3,3),dtype=tl.float64,device="cuda")``` after ```tl.set_backend('numpy')``` raises no warning/error, perhaps it should?

#### Versions
Windows-10-10.0.19045-SP0
Python 3.11.4 | packaged by Anaconda, Inc. | (main, Jul  5 2023, 13:38:37) [MSC v.1916 64 bit (AMD64)]
NumPy 1.26.4
SciPy 1.10.1
TensorLy 0.8.1
PyTorch 2.2.0+cu121

**Comment by JeanKossaifi** [comment]
Yes, currently the context of a tensor depends on the backend. I did think about making it backend agnostic, I guess that is one option. We then need to normalize the device across all backends then. 

We could introduce e.g. tl.cpu and tl.gpu if necessary. What do you think @aarmey @cohenjer @MarieRoald @yngvem ?

**Comment by yngvem** [comment]
I think we should wait a bit with that, NumPy will soon add support for the [array API](https://data-apis.org/array-api/latest/API_specification/array_object.html) in [NumPy v2](https://numpy.org/neps/nep-0047-array-api-standard.html#syntax-for-device-support) later this year.

**Comment by JeanKossaifi** [comment]
I agree @yngvem - actually really excited about the cleaned up API, it's amazing to have such a big project do such a large cleanup!

**Comment by JeanKossaifi** [comment]
Should we revisit this?

**Comment by pfisterj** [comment]
I have tested with numpy 2.0.0, pytorch 2.3.1 and tensorly 0.8.1. Now this code works like a charm:

```python
# works OK
tl.set_backend('pytorch')
A = tl.tensor((3,3),dtype=tl.float32)
print("torch|A:",A)
print("torch|A.dtype:",A.dtype)
print("torch|A.device:",A.device)

# works OK
tl.set_backend('numpy')
A = tl.tensor((3,3),dtype=tl.float32,device="cpu")
print("numpy|A:",A)
print("numpy|A.dtype:",A.dtype)
print("numpy|A.device:",A.device)
```
Notice that I have considered `dtype=tl.float32`. When using `dtype=tl.float64` I get

```python
tl.set_backend('pytorch')
print(tl.float32)
>>> torch.float32
print(tl.float64)
>>> <class 'numpy.float64'>
A = tl.tensor((3,3),dtype=tl.float64)
>>> ValueError: invalid type: 'numpy.float64'
```
Did I miss something here?

**Comment by aarmey** [comment]
NumPy has deprecated `np.float64` in favor of the built-in `float` type. We probably just need to redefine `tl.float64` for the NumPy backend.
